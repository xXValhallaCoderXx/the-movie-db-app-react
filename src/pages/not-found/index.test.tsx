import "@testing-library/jest-dom/extend-expect";
import React from "react";
import {render, fireEvent} from "@testing-library/react";
import {StaticRouter} from "react-router";
import NotFoundPage from "./index";

test("Not found page renders with expected content", () => {
  const message1 = "Ooops! Looks like you shouldn't be here!";
  const message2 = "Click HERE to go back";
  const {queryByText, getByLabelText, getByText, getByTestId} = render(
    <StaticRouter>
      <NotFoundPage />
    </StaticRouter>
  );
  expect(getByText(message1)).toBeInTheDocument();
  // expect(getByText(message2)).toBeInTheDocument();
});

test("Redirected from Not Found page on link click", () => {
  const message1 = "Ooops! Looks like you shouldn't be here!";
  const {getByTestId, getByText} = render(
    <StaticRouter>
      <NotFoundPage />
    </StaticRouter>
  );
  const redirectLink = getByTestId("home-link");
  fireEvent.click(redirectLink);
  expect(getByText(message1)).not.toBeInTheDocument();
});
