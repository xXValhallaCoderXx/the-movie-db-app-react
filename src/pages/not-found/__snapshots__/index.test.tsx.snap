// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page should render correctly 1`] = `
<DefaultLayout>
  <div
    className="bg-tmd-dark-green h-screen flex justify-center items-center"
  >
    <div
      className="max-w-sm rounded overflow-hidden shadow-lg bg-white p-10"
    >
      <h1
        className="text-2xl text-center"
      >
        Ooops! Looks like you shouldn't be here!
      </h1>
      <p
        className="text-center mt-5"
      >
        Click
         
        <span
          className="font-bold"
        >
          <Link
            data-testid="home-link"
            to="/movies"
          >
            HERE
          </Link>
        </span>
         
        to go back
      </p>
    </div>
  </div>
</DefaultLayout>
`;
